apps/dashboard/src/utils/liquid-autocomplete.tsx

```diff
@@ -281,7 +292,23 @@ function getMatchingVariables(
     return acc;
   }, []);
 
-  const baseVariables = Array.from(new Map([...jitVariables, ...variables].map((item) => [item.name, item])).values());
+  const prefix = searchText.split('.')[0];
+  const localVariable = scopedVariables.find((v) => v.name === prefix);
+  let combinedVariables = [...jitVariables, ...allVariables];
+
+  if (localVariable) {
+    combinedVariables = [
+      ...jitVariables,
+      {
+        name: searchText,
+        displayLabel: searchText,
+        type: 'local',
+      },
+      ...allVariables,
+    ];
+  }
```

Comment:
if we found the local variable for the prefix that the user typed, for example `product.name` where `product` is prefix, then the `product.name` will be added to the suggestions list as it's valid local variable